src/site/content/en/fast/efficiently-load-third-party-javascript/index.md

```diff
@@ -0,0 +1,174 @@
+
+{% Aside 'caution' %}
+
+Be careful when lazy-loading resources with JavaScript. If JavaScript fails to load, perhaps due to flaky network conditions, your resources won’t load at all. Keep an eye out for [native lazy-loading support for images and iframes](https://addyosmani.com/blog/lazy-loading/) that’s coming soon to Chrome.
+
+{% endAside %}
+
+## Optimize how you serve third-party scripts
+
+
+### Use third-party CDN hosting
+
+It’s common for third-party vendors to provide URLs of JavaScript files that they host, usually on a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network). The benefits of this approach are that you can get started quickly—just copy and paste the URL—and there’s no maintenance overhead. The third-party vendor handles server configuration and script updates.
+
+But because they are not on the same origin as the rest of your resources, loading files from a public CDN comes with a network cost. The browser needs to perform a DNS lookup, establish a new HTTP connection, and, on secure origins, perform an SSL handshake with the vendor’s server. 
+
+When you use files from third-party servers, you rarely have control over caching. Relying on someone else’s caching strategy might cause scripts to be unnecessarily re-fetched from the network too often.
+
+{% Aside %}
+
+One possible benefit of serving third-party code this way is cross-domain caching. Sometimes a resource served from a public CDN may already be cached in the user’s browser if they previously visited another site that used the same resource from the same public CDN. This optimization[ doesn’t happen in Safari](https://andydavies.me/blog/2018/09/06/safari-caching-and-3rd-party-resources/) due to security reasons and it’s hard to tell how common it is in other browsers.
+{% endAside %}
+
+### Self-host third-party scripts
+
+Self-hosting third-party scripts is an option that gives you more control over a script’s loading process. By self-hosting you can: 
+
+* Reduce DNS lookup and round-trip times. 
+* Improve [HTTP caching](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) headers.
+* Take advantage of [HTTP/2 server push](https://developers.google.com/web/fundamentals/performance/http2/). 
+
+For example, Casper managed to [shave 1.7 seconds](https://medium.com/caspertechteam/we-shaved-1-7-seconds-off-casper-com-by-self-hosting-optimizely-2704bcbff8ec) off load time by self-hosting an A/B testing script.
+
+Self-hosting comes with one big downside though: scripts can go out of date and won’t get automatic updates when there’s an API change or a security fix. 
```

Comment:
Can we note a way to mitigate this issue? Like, say, package management?

Reply:
cc @addyosmani #phoneafriend